### DISCUSSION TITLE
Need help adding a language (Hokkien)

### DISCUSSION THREAD

**Discussion by hiong3-eng5** [espeak-ng/espeak-ng#1477] [Q&A] [answered]
I am trying to add my language Hokkien {a Language is China's Fujian Province). Built espeak-ng successfully using make -B.
Added all the files needed for phonemes and dictionary list and rules. Added entries to files phonemes and Makefile.am.

tried to create the hokkien phoneme via the command '`espeak-ng --compile=hbl`', but got an error message:'`The specified espeak-ng voice does not exist`'.

while the command `espeak-ng -compile-phonemes=phsource` while in the root directory `espeak-ng`. produced

```
Compiling phoneme data: phsource/phonemes
# This file lists the type of data that has been compiled into the
# phondata file
#
# The first character of a line indicates the type of data:
#   S - A SPECT_SEQ structure
#   W - A wavefile segment
#   E - An envelope
#
# Address is the displacement within phondata of this item
#
#  Address  Data file
#  -------  ---------

```
This indicates that the Hokkien phoneme was not added right?

Did I miss anything? All the files is located in the espeak-ng's git clone, in their respective directories.

Thanks.

**Comment by jaacoppi** [comment, marked as answer]
Thank you for the new language!

You don't need to use compile or compile-phonemes manually. Make will do it automatically if everything is setup correctly. The file espeak-ng-data/hbl_dict should appear.

Can you show your source code? Link to github is preferable but a diff patch works as well.


If you want to do it manually, check ESPEAK_DATA_PATH and LD_LIBRARY_PATH. This should work:
cd dictsource && ESPEAK_DATA_PATH=`pwd`/.. LD_LIBRARY_PATH=src:${LD_LIBRARY_PATH}/.. ../src/espeak-ng --compile=hbl

**Reply by hiong3-eng5** [reply]
Thanks for the reply, I'll try your suggestions. Again, thanks.

**Reply by hiong3-eng5** [reply]
> Thank you for the new language!
> 
> You don't need to use compile or compile-phonemes manually. Make will do it automatically if everything is setup correctly. The file espeak-ng-data/hbl_dict should appear.
> 
> Can you show your source code? Link to github is preferable but a diff patch works as well.
> 
> 
> If you want to do it manually, check ESPEAK_DATA_PATH and LD_LIBRARY_PATH. This should work:
> cd dictsource && ESPEAK_DATA_PATH=`pwd`/.. LD_LIBRARY_PATH=src:${LD_LIBRARY_PATH}/.. ../src/espeak-ng --compile=hbl
> 

Thank you so much, I found my bug. And learned a lot from the paths, and suggestion. Thanks, I can now test the language.

Thank you so much!
